### DISCUSSION TITLE
Can this run in a docker container?

### DISCUSSION THREAD

**Discussion by caldenmic** [jhuckaby/Cronicle#818] [Q&A]
I understand that if this runs as a docker container that the schedules will run inside the container rather that on the host? Is this a valid question or can I run it in a docker container?

**Comment by jhuckaby** [comment]
Yes, just select the correct target server (your container) to run your events on.

Check out Mike's fork for Docker installation instructions: https://github.com/cronicle-edge/cronicle-edge#docker

**Reply by caldenmic** [reply]
Thanks for guiding me is there a way I can uninstall this in my system after testing? I couldn't find it in the documentation

**Reply by jhuckaby** [reply]
You're absolutely right, that is missing from the docs.  So sorry, I'll add that.  It's super easy to uninstall.  Just stop the service and delete the `/opt/cronicle` directory:

```sh
/opt/cronicle/bin/control.sh stop
rm -rf /opt/cronicle
```

**Reply by caldenmic** [reply]
I wanted to clarify one other thing:

I have ran the docker containers in both machines lets say X (manager) and Y(worker). When I try to create a job and run it on Y from X it says:
```
Error: Failed to launch event: server or server group not found
```

I have added Y's IP address as a server but will this work?
But there after I add the IP address in the text box it says server added but in the table it shows the IP address of X instead of Y.

Below are the respective docker commands for manager and worker

Manager:
docker run -d -it --name cronicle-manager -p 3012:3012 ghcr.io/cronicle-edge/cronicle-edge:main manager
Worker:
docker run -d -it --name cronicle-edge -p 3012:3012 ghcr.io/cronicle-edge/cronicle-edge:main worker

**Reply by jhuckaby** [reply]
I do apologize, but this is beyond my expertise.  Try posting this issue over in Mike's fork and see if he or the community can help there: https://github.com/cronicle-edge/cronicle-edge/issues

**Reply by caldenmic** [reply]
Thanks for the help @jhuckaby its a really helpful tool

**Reply by caldenmic** [reply]
Hi I wanted to know how to add servers whenever I try to add a remote server with IP address the server adds and it shows a different IP address that starts with 172. The remote server is a azure VM where slave is running and im trying to add that IP address in the manager using add server

**Reply by jhuckaby** [reply]
Cronicle tries to guess what the server's IP address is, by reading the list of network interfaces on the machine, and picking the first one marked non-internal.  If your server has multiple external network interfaces, this may pick the wrong one.  To fix this, you can add the server's IP that you want Cronicle to use in the `/opt/cronicle/conf/config.json` file.  Add a top-level property named `ip` like this:

```
"ip": "1.2.3.4"
```
